src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,305 @@
+     *
+     *  After this text, the suggestion will be surrounded in quotes
+     *
+     * **Default value:** `Product recommandation for`
+     */
+    headerText: ComponentOptions.buildLocalizedStringOption({ defaultValue: 'ProductRecommandation' }),
+    /**
+     *  The time in `milliseconds` that a end user have to hover on a query suggest before a request is sent.
+     *
+     * **Default value:** `200`
+     */
+    hoverTime: ComponentOptions.buildNumberOption({ defaultValue: 200 })
   };
 
-  constructor(public element: HTMLElement, public options?: IQuerySuggestPreview, public bindings?: IComponentBindings) {
-    super(element, options, bindings, QuerySuggestPreview.ID);
+  private previousSuggestionHovered: string;
+  private renderer: ResultListRenderer;
+  private timer;
 
-    //Have to do this for now or or break the search interface
-    //Will fix in JSUI-2508 where i will use my template
-    this.element.style.display = 'none';
+  /**
+   * Creates a new QuerySuggestPreview component.
+   * @param element The HTMLElement on which to instantiate the component.
+   * @param options The options for the QuerySuggestPreview component.
+   * @param bindings The bindings that the component requires to function normally. If not set, these will be
+   * automatically resolved (with a slower execution time).
+   */
+  constructor(public element: HTMLElement, public options?: IQuerySuggestPreview, public bindings?: IComponentBindings) {
+    super(element, QuerySuggestPreview.ID, bindings);
 
     this.options = ComponentOptions.initComponentOptions(element, QuerySuggestPreview, options);
 
+    if (!this.options.resultTemplate) {
+      this.logger.warn(
+        `Option "resultTemplate" is *REQUIRED* on the component "QuerySuggestPreview". The component or the search page might *NOT WORK PROPERLY*`,
+        `Check the following URL to create your result template`,
+        `https://developers.coveo.com/x/aIGfAQ`
+      );
+    }
+
     this.bind.onRootElement(OmniboxEvents.querySuggestGetFocus, (args: IQuerySuggestSelection) => this.querySuggestGetFocus(args));
   }
 
-  private get shouldShowPreviewResults() {
-    return this.options.numberOfPreviewResults > 0;
+  /**
+   * Builds and returns an array of HTMLElement with the given result set.
+   * @param results the result set to build an array of HTMLElement from.
+   */
+  public buildResults(results: IQueryResults): Promise<HTMLElement[]> {
+    const res: { elem: HTMLElement; idx: number }[] = [];
+    const resultsPromises = map(results.results, (result: IQueryResult, index: number) => {
+      return this.buildResult(result).then((resultElement: HTMLElement) => {
+        if (resultElement != null) {
+          res.push({ elem: resultElement, idx: index });
+        }
+        return resultElement;
+      });
+    });
+
+    // We need to sort by the original index order, because in lazy loading mode, it's possible that results does not gets rendered
```

Comment:
So, this whole function and the next after that is kind of a big copy paste from what's already in `ResultList.ts` :)

The correct way to fix that will be to think about the "logical" unit or piece of code that could be extracted between the two components, and that could live as a standalone class/module.

What I mean by "standalone" is that it should be testable on it's own and agnostic from ResultList/QuerySuggest preview.

Then, import and compose both components reusing that shared piece of code.